src/main/scala/piconot/external/parser/Parser.scala

```diff
@@ -0,0 +1,67 @@
+package piconot.external.parser
+
+import scala.util.parsing.combinator._
+import piconot.external._
+import picolib.semantics._
+
+/**
+ * Parser for external DSL
+ */
+object PicoParser extends JavaTokenParsers with PackratParsers {
+  
+  /**
+   * Allows for parsing comments
+   */
+  override protected val whiteSpace = """(\s|#.*)+""".r
+  
+  /**
+   * Applies the parser to a string
+   */
+  def apply(s: String): ParseResult[List[StateStruct]] = parseAll(program,s)
+  
+  /**
+   * A program can be made of any number of StateStructs
+   */
+  def program: PackratParser[List[StateStruct]] = struct*
+  
+  /**
+   * A StateStruct has the state name and the a list of PRules
+   */
+  def struct: PackratParser[StateStruct] =
+    "State" ~ state ~ ":" ~ prules ^^
+      {case "State" ~ n ~ ":" ~ r => StateStruct(n,r)}
+  
+  /**
+   * A list of PRules is made of one or more PRule
+   */
+  def prules: PackratParser[List[PartialRule]] = prule*
+  
+  /**
+   * A prule is a comma separated blocked and opened arrows followed by a '=>' followed by comma separated
+   *  move direction and next state
+   */
+  def prule: PackratParser[PartialRule] =
+    arrows ~ "," ~ arrows ~ "=>" ~ arrow ~ "," ~ state ^^
+      {case b ~ "," ~ o ~ "=>" ~ m ~ "," ~ s => PartialRule(b,o,m,s)}
+  
+  /**
+   * Any number of arrows mapped to MoveDirections
+   */
+  def arrows: PackratParser[List[MoveDirection]] = arrow*
+  
+  /**
+   * Maps an arrow to a MoveDirection
+   */
+  def arrow: PackratParser[MoveDirection] = (
+        ("↑" ^^^ North)
+      | ("→" ^^^ East)
+      | ("←" ^^^ West)
+      | ("↓" ^^^ South)
+      | ("." ^^^ StayHere)
+   )
+   
+  /**
+   * Parses a state name
+   */
+  def state: PackratParser[State] = """\d\d?""".r ^^ State
+}
```

Comment:
I think this is straightforward and really the right way about this. It's similar to what we had to do -- the approach ends up being completely different, and you have far fewer possible strings/combinations to parse, so the end result is completely different. This is probably nicer than what we did. I like the use of plural words leading to a single rule with *; helps a lot with clarity.